moved the calls back

rosenhauer opened the pull request:
I moved the calls for lidHookOutside();   buildLightTubes(); buildButtons(); lidHookInside(); back to the original locations.  This fixed the button issue.

But it does break (well maybe not break but change) the inspectx/y functionality. But I do see that you had separate inspect options for the light tubes and buttons so having them as part of the inspect function wasn't there before.  If I get a chance I may look at how to get the inspectx/y working for everything but as you have the other options it's probably not necessary.

The inside/outside functions do not work still.  If I have a chance to look at that more later I will.  

Will probably look at adding fillets to the button recess also. 
![image](https://github.com/mrWheel/YAPP_Box/assets/33765928/03362db1-40ae-4866-97f0-085ec737a05e)
![image](https://github.com/mrWheel/YAPP_Box/assets/33765928/a49c138f-8a98-48b3-8ddb-c30647ce9e18)

mrWheel commented:
@rosenhauer,

I restored a working v2.0.5 version of `YAPPgenerator_v20.scad` so the test and example programs work with that version (basically removed fillets from that version).
Also added test programs for v2.1

Now investigating how to add a v2.1 tag to this new version... -done that before but no idea how :-(

Hope this does not interfere with your (future) work.

Regards

rosenhauer commented:
First off thanks for creating this project.

I'm not wanting to take it over as it's your baby, but as I have my first
3D printer on order and already have 4 different project boxes to make I'm
running into limitations that I can help eliminate.

I've been going through the code more closely and found some
inconsistencies and things that I think would improve the code.

Unless you have an objection I'm working on the following:

1. Update the debug inspect code.
  a. I've moved it in my test copy to the very end (after everything is
rendered) so one config will cut everything.
  b. I changed it to be from the true origin and configurable front/back
left/right so you can even cut it outside the main body and through the
buttons.

2. I already added an option for cutting the cutouts to only the thickness
of the face it's on (unfortunately I don't think I had that all working in
the copy I uploaded last time). This is helpful if you have connectors/pcb
supports to avoid below.
[image: image.png]
  a. I was thinking of changing these to have a depth parameter with 0
defaulting to the face thickness.
  b. Deeper cuts can be used to create vents that wrap around the edges of
the box.
[image: image.png]
  c. I also noticed that how you were calculating the cutouts is different
for different sides (like the right/left differ by more than just the shift
to the different plane so I was going to rewrite that into a single module
that you can specify the plane in.
  d. I also noticed that you have the cutouts for the screw holes in
the cutoutsInXY so they are actually cut out multiple times (not a problem
in the final result but takes more time). I will separate those out also.

3. Some more options to have preview and render quality options so you can
get fast onscreen and higher quality final renderings.

4. Making a "master" config embedded in the generator that you can enable
with a keyword to make sure that every combination of options can be tested.
  a. It will be a rather large box with an example of each feature:
    Cutouts on each face
    Base Mounts
    Snap connectors
    Light Tubes
    Buttons
    Connectors
  b. Every option for each of the above. for example:
    Shape (square, round, user defined)
    each coordinate system
    with and without rotation

5. I see that there are a few different coordinate options:
//-- pcb_standoffs  -- origin is pcb[0,0,0]
//-- base plane    -- origin is pcb[0,0,0]
//-- Lid plane    -- origin is pcb[0,0,0]
//-- cutoutsGrill    -- origin is pcb[0,0,0]
//-- front plane  -- origin is pcb[0,0,0]
//-- back plane  -- origin is pcb[0,0,0]
//-- left plane   -- origin is pcb[0,0,0]
//-- right plane   -- origin is pcb[0,0,0]
//-- connectors
  //-- normal         : origen = box[0,0,0]
  //-- yappConnWithPCB: origen = pcb[0,0,0]
//-- base mounts -- origen = box[x0,y0]
//-- snap Joins -- origen = box[x0,y0]
//-- origin of labels is box [0,0,0]
//-- lightTubes  -- origin is pcb[0,0,0]
//-- pushButtons  -- origin is pcb[0,0,0]
//-- origin of labels is box [0,0,0]

Several of my projects have multiple circuit boards so for many of the
options like cutouts I would prefer to use box coordinates.  So I was
thinking of adding an optional coordinate system to the ones that it makes
sense to.

6. You also have yappCenter options for most (but not all objects - for
example circular cutouts don't have yappCenter code/options) and it looks
like yappCenter isn't consistent in how it's implemented.  In some cases
it's done by adjusting the parameters for example in makeCutoutsInXZ and
other places it uses the "center = true" option for example in
cutoutsInXY.  The latter adjusts all three axis and the former is just
adjusting the axis that the position specifies. If I combine all of the
cutouts into a single method it will make them all implemented the same.


Unless you are opposed to any of these changes I'll work on them over the
next little bit.  My printer should be here at the end of the week. So I'm
trying to get my designs done before then so I have something useful
to test out the printer with.

Regards
Dave

On Tue, Nov 14, 2023 at 4:55 AM mrWheel ***@***.***> wrote:

> @rosenhauer <https://github.com/rosenhauer>,
>
> I restored a working v2.0.5 version of YAPPgenerator_v20.scad so the test
> and example programs work with that version (basically removed fillets from
> that version).
> Also added test programs for v2.1
>
> Now investigating how to add a v2.1 tag to this new version... -done that
> before but no idea how :-(
>
> Hope this does not interfere with your (future) work.
>
> Regards
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mrWheel/YAPP_Box/pull/25#issuecomment-1809982434>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIBTUKBVA2OE6NYTOQQD3UTYENE33AVCNFSM6AAAAAA7JKBL4KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMBZHE4DENBTGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Dave

mrWheel commented:
@rosenhauer,

Yes, it ís my “baby” but as I wrote before: I’m an incremental developer and after a view iterations of the code I’m losing sight of what I have coded already and sometimes I reinvent modules I already have.

So your findings after the code review are correct. There is some inconsistency in the API’s for the deferent array’s.

If you have the time please make the code better and more consistent!
The YAPPgenerator is used by a lot of people and they will all benefit from your work.

What printer have you bought?

Regards,

Willem

rosenhauer commented:
Sounds like a plan.  Making it better for everyone was my goal.  I'm a
full-time developer but enjoy helping out on projects like this on the
side.

I got a refurbished Ender 3 S1.  I think it will be good for my first one.

On Wed, Nov 15, 2023, 1:18 AM mrWheel ***@***.***> wrote:

> @rosenhauer <https://github.com/rosenhauer>,
>
> Yes, it ís my “baby” but as I wrote before: I’m an incremental developer
> and after a view iterations of the code I’m losing sight of what I have
> coded already and sometimes I reinvent modules I already have.
>
> So your findings after the code review are correct. There is some
> inconsistency in the API’s for the deferent array’s.
>
> If you have the time please make the code better and more consistent!
> The YAPPgenerator is used by a lot of people and they will all benefit
> from your work.
>
> What printer have you bought?
>
> Regards,
>
> Willem
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mrWheel/YAPP_Box/pull/25#issuecomment-1811927514>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIBTUKDGEDUUEKH3YBLCK6TYERUERAVCNFSM6AAAAAA7JKBL4KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMJRHEZDONJRGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
